reactor-test/src/test/java/reactor/test/publisher/ColdTestPublisherTests.java

```diff
@@ -111,6 +124,21 @@ public void coldDisallowsOverflow() {
 		publisher.assertNoRequestOverflow();
 	}
 
+	@Test
+	public void misbehavingAllowsOverflow() {
+		TestPublisher<String> publisher = TestPublisher.createColdNonCompliant(false, TestPublisher.Violation.REQUEST_OVERFLOW);
+
+		assertThatExceptionOfType(AssertionError.class)
```

Comment:
If that test is made that way because the `StepVerifier` complains that its under-requesting when adding an `expectNext("bar")`, that under-requesting check can be disabled using `StepVerifierOptions` (namely, `StepVerifierOptions#checkUnderRequesting(false)`).

Reply:
I'm not sure I get what you mean by "made that way" (do you mean using `assertThatExceptionOfType().isThrownBy()` (ie direct exception)?).
That test was copied and adapted from the `DefaultTestPublisher` IIRC.

Reply:
after looking at the test again, it appears it is fine, nevermind